include/lexer.h

```diff
@@ -0,0 +1,25 @@
+#ifndef LEXER_H
+#define LEXER_H
+
+#include <stdbool.h>
+
+#include "charstream.h"
+#include "token.h"
+
+struct quoting_state {
+  bool backslashed;
+};
+
+struct lexer {
+  struct charstream stream;
+  struct quoting_state quoting;
+
+  struct token *cur;
+};
+
+int lexer_init(struct lexer *l, FILE *input);
+void lexer_cleanup(struct lexer *l);
+
+const struct token *lexer_read(struct lexer *l);
```

Comment:
``lexer_get_token``? ``lexer_next_token``? Not a fan of ``lexer_read``

Reply:
What about `lexer_consume` or `lexer_take`? I agree that `lexer_read` isn't the best, as it doesn't show that the lexer consumes the input